### PR TITLE
Switch to use CW add-on for Python EKS E2E test.

### PR DESCRIPTION
**Pull request by zzhlogin**
Switch to use CW add-on for Python EKS E2E test. Adding back Java E2E test trace validation.
Related change in test framework repo: 
https://github.com/aws-observability/aws-application-signals-test-framework/pull/32

Test workflow run:
https://github.com/aws/amazon-cloudwatch-agent-operator/actions/runs/8527918991

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by lisguo — APPROVED** [review]

File: .github/workflows/appsignals-python-e2e-test.yml [hunk 1]
```diff
@@ -147,31 +125,36 @@ jobs:
...
-              cut -d'/' -f2
-            )
-            aws iam attach-role-policy --role-name $node_role --policy-arn arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy --region ${{ env.AWS_DEFAULT_REGION }}
-            aws iam attach-role-policy --role-name $node_role --policy-arn arn:aws:iam::aws:policy/AWSXRayWriteOnlyAccess --region ${{ env.AWS_DEFAULT_REGION }}
-          done
+          kubectl get pods -n ${{ env.SAMPLE_APP_NAMESPACE }}
+
+      - name: Wait for sample app pods to come up
+        run: |
+          kubectl wait --for=condition=Ready pod --all -n ${{ env.SAMPLE_APP_NAMESPACE }} \
```
Comment (majanjua-amzn) [inline .github/workflows/appsignals-python-e2e-test.yml:157]: No need for this to be an entirely separate step from the previous one

File: .github/workflows/appsignals-python-e2e-test.yml [hunk 2]
```diff
@@ -31,20 +31,31 @@ env:
...
+            scripts/eks/appsignals/clean-app-signals.sh
+          sparse-checkout-cone-mode: false
+
+      - name: Use latest addon version
+        working-directory: enablement-script/scripts/eks/appsignals
+        run: |
+          sed -i 's/aws eks create-addon \\/aws eks create-addon \\\n        --resolve-conflicts OVERWRITE \\/' enable-app-signals.sh
+          sed -i 's/if \[\[ "$addon_version" < "v1\.2\.0" \]\]/if [[ "$addon_version" < "v1.4.0" ]]/' enable-app-signals.sh
+          sed -i 's/--addon-version v1\.2\.0-eksbuild\.1/--addon-version v1.4.0-eksbuild.1/' enable-app-signals.sh
+          cat enable-app-signals.sh
```
Comment (majanjua-amzn) [inline .github/workflows/appsignals-python-e2e-test.yml:58]: Same comment as in https://github.com/aws-observability/aws-otel-python-instrumentation/pull/124

File: .github/workflows/appsignals-python-e2e-test.yml [hunk 3]
```diff
@@ -31,20 +31,31 @@ env:
   APP_SIGNALS_PYTHON_E2E_RE_SA_IMG: appsignals-python-django-remote-service
   METRIC_NAMESPACE: AppSignals
   LOG_GROUP: /aws/appsignals/eks
-#  ECR_OPERATOR_STAGING_REPO: ${{ vars.ECR_OPERATOR_STAGING_REPO }}
+  ECR_OPERATOR_STAGING_REPO: ${{ vars.ECR_OPERATOR_STAGING_REPO }}
```
Comment (majanjua-amzn) [inline .github/workflows/appsignals-python-e2e-test.yml:34]: What's `vars`? Would we want to use secrets instead?

Comment (zzhlogin) [reply .github/workflows/appsignals-python-e2e-test.yml:34]: This is referencing to `ECR_OPERATOR_STAGING_REPO` variable, which is used in ["Build And Upload Artifact" workflow](https://github.com/aws/amazon-cloudwatch-agent-operator/blob/8472f520bd50707494064564d33658b231518df0/.github/workflows/build-and-upload-release.yml#L8), the staging artifacts will be saved in this repo. So, I'm using the same variable here to make sure the image can be found.

File: .github/workflows/appsignals-python-e2e-test.yml (hunk 2 above)
Comment (zzhlogin) [reply .github/workflows/appsignals-python-e2e-test.yml:58]: Updated.

File: .github/workflows/appsignals-python-e2e-test.yml (hunk 1 above)
Comment (zzhlogin) [reply .github/workflows/appsignals-python-e2e-test.yml:157]: Updated.

File: .github/workflows/appsignals-python-e2e-test.yml [hunk 4]
```diff
@@ -31,20 +31,28 @@ env:
...
-          fetch-depth: 0
+          repository: aws-observability/application-signals-demo
+          ref: main
+          path: enablement-script
+          sparse-checkout: |
+            scripts/eks/appsignals/enable-app-signals.sh
+            scripts/eks/appsignals/clean-app-signals.sh
+          sparse-checkout-cone-mode: false
+
+      - name: Resolve Add-on configuration conflict
```
Comment (majanjua-amzn) [inline .github/workflows/appsignals-python-e2e-test.yml:52]: nit: Should add a comment about why this is needed.

**Review by majanjua-amzn — APPROVED** [review]

**Review by majanjua-amzn — APPROVED** [review]

**Review by lisguo — APPROVED** [review]

**Pull request merged**